components/Typography/Text.tsx

```diff
@@ -26,7 +26,8 @@ export type TypographyProps = {
   children?: ReactNode;
 };
 
-export type TextProps = TypographyProps & React.HTMLAttributes<HTMLElement>;
+// The TimeHTMLAttributes<HTMLElement> is for the dateTime attribute when using as="time"
+export type TextProps = TypographyProps & React.HTMLAttributes<HTMLElement> & TimeHTMLAttributes<HTMLElement>;
```

Comment:
For some reason the dateTime attribute in the html `<time>` isn't included in `React.HTMLAttributes<HTMLElement>` so added additional `TimeHTMLAttributes<HTMLElement>`